src/proposal/plist.mligo

```diff
@@ -57,9 +57,8 @@ let plist_insert (key, plist_o : proposal_key * (proposal_doubly_linked_list opt
 let plist_delete (key, plist_o : proposal_key * proposal_doubly_linked_list option): proposal_doubly_linked_list option =
   match plist_o with
     | None ->
-        // Expect to have at least a proposal to delete it.
-        ([%Michelson ({| { FAILWITH } |} : (nat * string) -> proposal_doubly_linked_list option)]
-            (bad_state, "no proposal") : proposal_doubly_linked_list option)
+        // Do nothing in case of no proposal key.
+        plist_o
```

Comment:
Also, why was this required? I mean, Is this meant to be reverted during merge?

Reply:
Most delete func (Data.List.delete, Map.delete) does not throw error when the key does not exist. So I want to keep the same functionality here (It is an impossible state anyway, since we use plist_delete on key that we're sure it exists).

Reply:
I think when we test something, we need to isolate the entity that is being tested, and define and isolate the behavior that is being tested, and write tests for that behavior. And I think we should not consider how some other code will be using the code under test, when writing tests, becaue that external code might use it in a different way in future.

So I think in these tests, we should document the expected behavior of plist. If it is close to the behavior of data structure X, we can say that plist behave mostly like X, except for so and so cases, and that is the behavior that we are testing. 

This is also regarding this thread https://github.com/tezos-commons/baseDAO/pull/320/files#r819505802 regarding the insertion behavior.

Reply:
The closest data structure to plist is a unique element list (or a set that is ordered by its insertion time). Also if the same element is inserted it shouldn't overwrite the existing element. This is the expected behavior, I'll add this as a comments.